src/System Application/App/Email/src/Email/Email.Codeunit.al

```diff
@@ -196,6 +196,96 @@ codeunit 8901 Email
 
     #endregion
 
+    #region Reply
+
+    /// <summary>
+    /// Sends a reply to an external message id in the foreground.
+    /// </summary>
+    /// <param name="EmailMessage">The email message with the details of the recipients and reply to be added.</param>
+    /// <param name="ExternalId">The external message id that is used to correlate and send the reply.</param>
+    /// <param name="EmailAccountId">The ID of the email account to use for sending the email.</param>
+    /// <param name="EmailConnector">The email connector to use for sending the email.</param>
+    /// <param name="EmailOutbox">The email outbox which is set up for sending in the background.</param>
+    /// <returns>True if sent</returns>
+    procedure Reply(EmailMessage: Codeunit "Email Message"; ExternalId: Text; EmailAccountId: Guid; EmailConnector: Enum "Email Connector"): Boolean
+    begin
+        exit(EmailImpl.Reply(EmailMessage, ExternalId, EmailAccountId, EmailConnector));
+    end;
+
+    /// <summary>
+    /// Sends a reply to an external message id to all recipients on that email in the foreground.
+    /// </summary>
+    /// <param name="EmailMessage">The email message with the details of the recipients and reply to be added.</param>
+    /// <param name="ExternalId">The external message id that is used to correlate and send the reply.</param>
+    /// <param name="EmailAccountId">The ID of the email account to use for sending the email.</param>
+    /// <param name="EmailConnector">The email connector to use for sending the email.</param>
+    /// <param name="EmailOutbox">The email outbox which is set up for sending in the background.</param>
+    /// <returns>True if sent</returns>
+    procedure ReplyAll(EmailMessage: Codeunit "Email Message"; ExternalId: Text; EmailAccountId: Guid; EmailConnector: Enum "Email Connector"): Boolean
+    begin
+        exit(EmailImpl.Reply(EmailMessage, ExternalId, EmailAccountId, EmailConnector, true));
+    end;
+
+    /// <summary>
+    /// Sends a reply to an external message id in the background.
```

Comment:
have you double checked all the foreground/background calls and docs cuz it's not obvious from the code when the thing will do which...

Reply:
Yes. The ones with a `var EmailOutbox` are the background functions while the ones without aren't. I'll see if I can do something to make it more obvious.

Reply:
Trying to avoid putting a parameter, as that's pointless when the difference is the `EmailOutbox` parameter. Similarly to how we have `Enqueue` and `Send`. I could do `EnqueueReply`.